### ISSUE TITLE
Stepper has an illogical sequence for the expanded button

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3560]
## Describe the bug

The interstitial content in the 'clr-accordion-header' used in the Stepper component presents an illogical reading sequence for screen reader users. When screen reader users hear a button announce an expanded or collapsed state they will think that the button controls the readability of upcoming content. 

Discovering content related to the "clr-accordion-header-button" button immediately after hearing that it is in a collapsed state may be confusing. While this is a low severity issue when there is a small amount of interstitial content, the severity will increase in relation to the amount of interstitial content.

## How to reproduce

https://clarity.design/documentation/stepper

- Using a screen reader go to Stepper example
- Read one of the collapsed Stepper header buttons
- Observe that you read the value of the description immediately after reading the collapsed button. 

## Expected behavior

_A clear and concise description of what you expected to happen._

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

The issue is that the content that is controlled by the "clr-accordion-header-button" toggle button should be the next sibling of the toggle button. Having content that is not controlled by the toggle button as the next sibling will be confusing for screen reader users because it will seem like the toggle button does not work since it does not control the interstitial content.

**Comment by coryrylan** [comment]
Decision to move description into the button element, provide guidance on the amount of content within the header description.
